packages/react/scripts/buildIconTypes.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+
+const ICON_COMPONENT_DIRECTORY = path.join(
+  __dirname,
+  '..',
+  'src',
+  'components',
+  'Icon'
+);
+const ICONS_DIRECTORY = path.join(ICON_COMPONENT_DIRECTORY, 'icons');
+const ICON_TYPES_FILE = path.join(ICON_COMPONENT_DIRECTORY, 'types.ts');
+const ICONS_WITH_DIRECTIONS = [
+  'arrow',
+  'arrow-circle',
+  'chevron',
+  'chevron-double',
+  'triangle'
+];
+const DIRECTIONS = ['up', 'down', 'left', 'right'];
```

Comment:
In theory, any icon could have a direction since it's just a modifier. I don't have strong opinions here since these are all the icons that make sense to have directions, we would just need to remember to add them when adding new icons. 🤷